test/test_ast_generation.py

```diff
@@ -26,15 +28,15 @@ def read_python_source(path: str) -> str:
     return source
 
 
-def test_ast_generation_on_source_files(tmp_path: PurePath) -> None:
+@pytest.mark.parametrize("filename", PYTHON_SOURCE_FILENAMES)
```

Comment:
Thanks for the refactoring! However now we build the extension for each file, which is rather slow. Maybe we can use pytest's fixture machinery to create the extension once and pass it in to each test?

Reply:
That's probably the best way to go, yes!